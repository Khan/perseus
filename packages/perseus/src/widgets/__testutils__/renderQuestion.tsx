import {RenderStateRoot} from "@khanacademy/wonder-blocks-core";
import {render} from "@testing-library/react";
import * as React from "react";

import {
    testDependenciesV2,
    testDependencies,
    // eslint-disable-next-line import/no-relative-packages
} from "../../../../../testing/test-dependencies";
import {
    DependenciesContext,
    useDependencies,
    setDependencies,
} from "../../dependencies";
import * as Perseus from "../../index";
import {mockStrings} from "../../strings";
import {registerAllWidgetsForTesting} from "../../util/register-all-widgets-for-testing";

import type {APIOptions} from "../../types";
import type {PerseusRenderer} from "@khanacademy/perseus-core";
import type {PropsFor} from "@khanacademy/wonder-blocks-core";

type RenderResult = ReturnType<typeof render>;

type ExtraProps = Omit<PropsFor<typeof Perseus.Renderer>, "strings">;

export const renderQuestion = (
    question: PerseusRenderer,
    apiOptions: APIOptions = Object.freeze({}),
    extraProps?: ExtraProps,
): {
    container: HTMLElement;
    renderer: Perseus.Renderer;
    rerender: (question: PerseusRenderer, extraProps?: ExtraProps) => void;
    unmount: RenderResult["unmount"];
} => {
    setDependencies(testDependencies);
    registerAllWidgetsForTesting();

    let renderer: Perseus.Renderer | null = null;
    const {container, rerender, unmount} = render(
        <RenderStateRoot>
            <DependenciesContext.Provider value={testDependenciesV2}>
                <Renderer
                    ref={(node) => (renderer = node)}
                    question={question as any}
                    apiOptions={apiOptions}
                    extraProps={{
                        ...extraProps,
                        strings: mockStrings,
                    }}
                />
            </DependenciesContext.Provider>
        </RenderStateRoot>,
    );
    // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
    if (!renderer) {
        throw new Error(`Failed to render!`);
    }
    const renderAgain = (
        question: PerseusRenderer,
        extraProps?: ExtraProps,
    ) => {
        rerender(
            <RenderStateRoot>
                <DependenciesContext.Provider value={testDependenciesV2}>
                    <Renderer
                        ref={(node) => (renderer = node)}
                        question={question}
                        apiOptions={apiOptions}
                        extraProps={{
                            ...extraProps,
                            strings: mockStrings,
                        }}
                    />
                </DependenciesContext.Provider>
            </RenderStateRoot>,
        );
        if (!renderer) {
            throw new Error(`Failed to rerender!`);
        }
    };

    return {container, renderer, rerender: renderAgain, unmount};
};

const Renderer = React.forwardRef<
    Perseus.Renderer,
    {
        question: PerseusRenderer;
        apiOptions: APIOptions;
        extraProps?: PropsFor<typeof Perseus.Renderer>;
    }
>((props, ref) => {
    const dependencies = useDependencies();
    return (
        <Perseus.Renderer
            ref={ref}
            content={props.question.content}
            images={props.question.images}
            widgets={props.question.widgets}
            problemNum={0}
            apiOptions={props.apiOptions}
            strings={mockStrings}
            {...props.extraProps}
            {...dependencies}
        />
    );
});
