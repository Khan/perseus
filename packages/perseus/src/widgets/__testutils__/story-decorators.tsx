import * as React from "react";

export const mobileDecorator = (Story) => (
    <div className="framework-perseus perseus-mobile">
        <Story />
    </div>
);

export const articleDecorator = (Story) => (
    <div className="framework-perseus perseus-article">
        <Story />
    </div>
);

export const mobileArticleDecorator = (Story) => (
    <div className="framework-perseus perseus-mobile perseus-article">
        <Story />
    </div>
);

export const rtlDecorator = (Story) => (
    <div style={{direction: "rtl"}}>
        <Story />
    </div>
);
