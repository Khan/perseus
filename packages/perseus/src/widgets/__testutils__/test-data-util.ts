import type {PerseusItem, PerseusRenderer} from "@khanacademy/perseus-core";

/**
 * Creates a Perseus item with the provided question/PerseusRenderer data.
 * Useful for converting test data used with RendererWithDebugUI to test data
 * for ServerItemRendererWithDebugUI.
 *
 * @param {PerseusRenderer} question - The question to be included in the Perseus item.
 * @returns {PerseusItem} The created Perseus item.
 */
export const createPerseusItem = (question: PerseusRenderer): PerseusItem => {
    return {
        question: question,
        answer: null,
        itemDataVersion: {
            major: 0,
            minor: 1,
        },
        hints: [],
        answerArea: {
            calculator: false,
            chi2Table: false,
            financialCalculatorMonthlyPayment: false,
            financialCalculatorTotalAmount: false,
            financialCalculatorTimeToPayOff: false,
            periodicTable: false,
            periodicTableWithKey: false,
            tTable: false,
            zTable: false,
        },
    };
};
