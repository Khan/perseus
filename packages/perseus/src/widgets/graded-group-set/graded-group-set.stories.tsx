import {ArticleRendererWithDebugUI} from "../../../../../testing/article-renderer-with-debug-ui";

import {article1} from "./graded-group-set.testdata";

import type {Meta, StoryObj} from "@storybook/react";

const meta: Meta = {
    title: "Perseus/Widgets/Graded Group Set",
    component: ArticleRendererWithDebugUI,
};
export default meta;

type Story = StoryObj<typeof ArticleRendererWithDebugUI>;

export const Article1: Story = {
    args: {
        json: article1,
    },
};
