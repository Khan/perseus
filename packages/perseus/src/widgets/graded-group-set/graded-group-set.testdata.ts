import type {PerseusRenderer} from "@khanacademy/perseus-core";

export const article1: PerseusRenderer = {
    content:
        "#Section 1: Adding tenths less than one\n\n[[☃ graded-group-set 1]]\n\n\nBeautiful, let's move on to problems with whole numbers and tenths.",
    images: {},
    widgets: {
        "graded-group-set 1": {
            type: "graded-group-set",
            alignment: "default",
            static: false,
            graded: true,
            options: {
                gradedGroups: [
                    {
                        title: "Problem 1a",
                        content: "$0.5 + 0.4 =$   [[☃ numeric-input 1]]",
                        widgets: {
                            "numeric-input 1": {
                                type: "numeric-input",
                                alignment: "default",
                                static: false,
                                graded: true,
                                options: {
                                    static: false,
                                    answers: [
                                        {
                                            value: 0.9,
                                            status: "correct",
                                            message: "",
                                            simplify: "required",
                                            strict: false,
                                            maxError: null,
                                        },
                                    ],
                                    size: "normal",
                                    coefficient: false,
                                    labelText: "",
                                },
                                version: {major: 0, minor: 0},
                            },
                        },
                        images: {},
                        hasHint: true,
                        hint: {
                            content:
                                "There are many ways to solve this problem. Let's see two student solutions.\n\n###Student A's solution:\n\nI thought in terms of tenths.\n\n$\\phantom{=}0.5 + 0.4$\n\n$=5$ tenths $+ ~4$ tenths\n\n$=9$ tenths\n\n$=0.9$\n\n###Student B's solution:\n\nI used tenths grids.\n\n[[☃ image 1]]\n\n[[☃ image 2]]\n\n[[☃ image 3]]\n\n$\\blueD{0.5} + \\greenD{0.4} = 0.9$\n\n###The answer:\n\n$0.5 + 0.4 = 0.9$",
                            images: {},
                            widgets: {
                                "image 3": {
                                    type: "image",
                                    alignment: "block",
                                    static: false,
                                    graded: true,
                                    options: {
                                        static: false,
                                        title: "",
                                        range: [
                                            [0, 10],
                                            [0, 10],
                                        ],
                                        box: [234, 132],
                                        backgroundImage: {
                                            url: "web+graphie://ka-perseus-graphie.s3.amazonaws.com/cda00c119dac3e52c8ed150ef4a9a37355f5c713",
                                            width: 234,
                                            height: 132,
                                        },
                                        labels: [],
                                        alt: "A square divided into 10 rows to show tenths. 5 of the rows are shaded in blue and 4 of the rows are shaded in green.",
                                        caption: "",
                                    },
                                    version: {major: 0, minor: 0},
                                },
                                "image 2": {
                                    type: "image",
                                    alignment: "block",
                                    static: false,
                                    graded: true,
                                    options: {
                                        static: false,
                                        title: "",
                                        range: [
                                            [0, 10],
                                            [0, 10],
                                        ],
                                        box: [180, 132],
                                        backgroundImage: {
                                            url: "web+graphie://ka-perseus-graphie.s3.amazonaws.com/217d580bc0baddb903bbcb648fc8d3ea3d0f4408",
                                            width: 180,
                                            height: 132,
                                        },
                                        labels: [],
                                        alt: "A square divided into 10 rows to show tenths. 4 of the rows are shaded to represent 4 tenths.",
                                        caption: "",
                                    },
                                    version: {major: 0, minor: 0},
                                },
                                "image 1": {
                                    options: {
                                        static: false,
                                        title: "",
                                        range: [
                                            [0, 10],
                                            [0, 10],
                                        ],
                                        box: [180, 132],
                                        backgroundImage: {
                                            url: "web+graphie://ka-perseus-graphie.s3.amazonaws.com/2a56a60275b7227ed9c5b89e489587c8cb13eb7b",
                                            width: 180,
                                            height: 132,
                                        },
                                        labels: [],
                                        alt: "A square divided into 10 rows to show tenths. 5 of the rows are shaded to represent 5 tenths.",
                                        caption: "",
                                    },
                                    type: "image",
                                    version: {major: 0, minor: 0},
                                    graded: true,
                                    alignment: "block",
                                    static: false,
                                },
                            },
                        },
                        widgetEnabled: true,
                        immutableWidgets: false,
                    },
                    {
                        title: "Problem 1b",
                        content: "$0.6 + 0.4 =$   [[☃ numeric-input 1]]",
                        widgets: {
                            "numeric-input 1": {
                                type: "numeric-input",
                                alignment: "default",
                                static: false,
                                graded: true,
                                options: {
                                    static: false,
                                    answers: [
                                        {
                                            value: 1,
                                            status: "correct",
                                            message: "",
                                            simplify: "required",
                                            strict: false,
                                            maxError: null,
                                        },
                                    ],
                                    size: "normal",
                                    coefficient: false,
                                    labelText: "",
                                },
                                version: {major: 0, minor: 0},
                            },
                        },
                        images: {},
                        hasHint: true,
                        hint: {
                            content:
                                "There are many ways to solve this problem. Let's see two student solutions.\n\n###Student A's solution:\n\nI thought in terms of tenths.\n\n$\\phantom{=}0.6 + 0.4$\n\n$=6$ tenths $+ ~4$ tenths\n\n$=10$ tenths\n\n$=1$\n\n###Student B's solution:\n\nI used tenths grids.\n\n[[☃ image 1]]\n\n[[☃ image 2]]\n\n[[☃ image 3]]\n\n$\\blueD{0.6} + \\greenD{0.4} = 1$\n\n###The answer:\n\n$0.6 + 0.4 = 1$",
                            images: {},
                            widgets: {
                                "image 3": {
                                    type: "image",
                                    alignment: "block",
                                    static: false,
                                    graded: true,
                                    options: {
                                        static: false,
                                        title: "",
                                        range: [
                                            [0, 10],
                                            [0, 10],
                                        ],
                                        box: [234, 132],
                                        backgroundImage: {
                                            url: "web+graphie://ka-perseus-graphie.s3.amazonaws.com/54fbc06d92097119f1be5d1679391596209296a7",
                                            width: 234,
                                            height: 132,
                                        },
                                        labels: [],
                                        alt: "A square divided into 10 rows to show tenths. 6 of the rows are shaded in blue and 4 of the rows are shaded in green.",
                                        caption: "",
                                    },
                                    version: {major: 0, minor: 0},
                                },
                                "image 2": {
                                    type: "image",
                                    alignment: "block",
                                    static: false,
                                    graded: true,
                                    options: {
                                        static: false,
                                        title: "",
                                        range: [
                                            [0, 10],
                                            [0, 10],
                                        ],
                                        box: [180, 132],
                                        backgroundImage: {
                                            url: "web+graphie://ka-perseus-graphie.s3.amazonaws.com/217d580bc0baddb903bbcb648fc8d3ea3d0f4408",
                                            width: 180,
                                            height: 132,
                                        },
                                        labels: [],
                                        alt: "A square divided into 10 rows to show tenths. 4 of the rows are shaded to represent 4 tenths.",
                                        caption: "",
                                    },
                                    version: {major: 0, minor: 0},
                                },
                                "image 1": {
                                    options: {
                                        static: false,
                                        title: "",
                                        range: [
                                            [0, 10],
                                            [0, 10],
                                        ],
                                        box: [180, 132],
                                        backgroundImage: {
                                            url: "web+graphie://ka-perseus-graphie.s3.amazonaws.com/8e353e65446a4322f53d640aba33dccd69b0874c",
                                            width: 180,
                                            height: 132,
                                        },
                                        labels: [],
                                        alt: "A square divided into 10 rows to show tenths. 6 of the rows are shaded to represent 6 tenths.",
                                        caption: "",
                                    },
                                    type: "image",
                                    version: {major: 0, minor: 0},
                                    graded: true,
                                    alignment: "block",
                                    static: false,
                                },
                            },
                        },
                        widgetEnabled: true,
                        immutableWidgets: false,
                    },
                    {
                        title: "Problem 1c",
                        content: "$0.8 + 0.4 =$   [[☃ numeric-input 1]]",
                        widgets: {
                            "numeric-input 1": {
                                type: "numeric-input",
                                alignment: "default",
                                static: false,
                                graded: true,
                                options: {
                                    static: false,
                                    answers: [
                                        {
                                            value: 1.2,
                                            status: "correct",
                                            message: "",
                                            simplify: "required",
                                            strict: false,
                                            maxError: null,
                                        },
                                    ],
                                    size: "normal",
                                    coefficient: false,
                                    labelText: "",
                                },
                                version: {major: 0, minor: 0},
                            },
                        },
                        images: {},
                        hasHint: true,
                        hint: {
                            content:
                                "There are many ways to solve this problem. Let's see two student solutions.\n\n###Student A's solution:\n\nI thought in terms of tenths.\n\n$\\phantom{=}0.8 + 0.4$\n\n$=8$ tenths $+ ~4$ tenths\n\n$=12$ tenths\n\n$=1.2$\n\n###Student B's solution:\n\nI used tenths grids.\n\n[[☃ image 1]]\n\n[[☃ image 2]]\n\n[[☃ image 3]]\n$\\blueD{0.8} + \\greenD{0.4} = 1.2$\n\n###The answer:\n\n$ 0.8+0.4=1.2 $",
                            images: {},
                            widgets: {
                                "image 3": {
                                    type: "image",
                                    alignment: "block",
                                    static: false,
                                    graded: true,
                                    options: {
                                        static: false,
                                        title: "",
                                        range: [
                                            [0, 10],
                                            [0, 10],
                                        ],
                                        box: [354, 132],
                                        backgroundImage: {
                                            url: "web+graphie://ka-perseus-graphie.s3.amazonaws.com/4b595ea53e5edc2b5991f2171f384a7eae2eeb24",
                                            width: 354,
                                            height: 132,
                                        },
                                        labels: [],
                                        alt: "Two equal sized squares. Each square is divided into 10 rows to show tenths. In the first square, 8 of the rows are shaded in blue and 2 of the rows are shaded in green. In the second square, 2 of the rows are shaded in green.",
                                        caption: "",
                                    },
                                    version: {major: 0, minor: 0},
                                },
                                "image 2": {
                                    type: "image",
                                    alignment: "block",
                                    static: false,
                                    graded: true,
                                    options: {
                                        static: false,
                                        title: "",
                                        range: [
                                            [0, 10],
                                            [0, 10],
                                        ],
                                        box: [180, 132],
                                        backgroundImage: {
                                            url: "web+graphie://ka-perseus-graphie.s3.amazonaws.com/217d580bc0baddb903bbcb648fc8d3ea3d0f4408",
                                            width: 180,
                                            height: 132,
                                        },
                                        labels: [],
                                        alt: "A square divided into 10 rows to show tenths. 4 of the rows are shaded to represent 4 tenths.",
                                        caption: "",
                                    },
                                    version: {major: 0, minor: 0},
                                },
                                "image 1": {
                                    options: {
                                        static: false,
                                        title: "",
                                        range: [
                                            [0, 10],
                                            [0, 10],
                                        ],
                                        box: [180, 132],
                                        backgroundImage: {
                                            url: "web+graphie://ka-perseus-graphie.s3.amazonaws.com/58d09bc7bdfadbd860c55734bf146578f047edbc",
                                            width: 180,
                                            height: 132,
                                        },
                                        labels: [],
                                        alt: "A square divided into 10 rows to show tenths. 8 of the rows are shaded to represent 8 tenths.",
                                        caption: "",
                                    },
                                    type: "image",
                                    version: {major: 0, minor: 0},
                                    graded: true,
                                    alignment: "block",
                                    static: false,
                                },
                            },
                        },
                        widgetEnabled: true,
                        immutableWidgets: false,
                    },
                ],
            },
            version: {major: 0, minor: 0},
        },
    },
};

export const groupSetRadioRationaleQuestion: PerseusRenderer = {
    content:
        "#Section 1: Adding tenths less than one\n\n[[☃ graded-group-set 1]]\n\n\nBeautiful, let's move on to problems with whole numbers and tenths.",
    images: {},
    widgets: {
        "graded-group-set 1": {
            type: "graded-group-set",
            alignment: "default",
            static: false,
            graded: true,
            options: {
                gradedGroups: [
                    {
                        title: "Question 1",
                        content:
                            "Select the correct answer.\n\n[[\u2603 radio 1]]\n\n",
                        images: {},
                        widgets: {
                            "radio 1": {
                                graded: true,
                                version: {
                                    major: 1,
                                    minor: 0,
                                },
                                static: false,
                                type: "radio",
                                options: {
                                    choices: [
                                        {
                                            id: "0-0-0-0-0",
                                            content: "Incorrect",
                                            correct: false,
                                            rationale:
                                                "This is not the correct answer.",
                                        },
                                        {
                                            id: "1-1-1-1-1",
                                            content: "Incorrect",
                                            correct: false,
                                            rationale:
                                                "This is not the correct answer.",
                                        },
                                        {
                                            id: "2-2-2-2-2",
                                            content: "Correct",
                                            correct: true,
                                            isNoneOfTheAbove: false,
                                            rationale:
                                                "This is the correct answer.",
                                        },
                                        {
                                            id: "3-3-3-3-3",
                                            content: "Incorrect",
                                            correct: false,
                                            isNoneOfTheAbove: false,
                                            rationale:
                                                "This is not the correct answer.",
                                        },
                                    ],
                                    countChoices: false,
                                    hasNoneOfTheAbove: false,
                                    multipleSelect: false,
                                    randomize: false,
                                    deselectEnabled: false,
                                },
                                alignment: "default",
                            },
                        },
                        hint: {
                            content: "This is an example hint.",
                            images: {},
                            widgets: {},
                        },
                        widgetEnabled: true,
                        immutableWidgets: false,
                    },
                    {
                        title: "Question 2",
                        content:
                            "What are some ways to say hello?\n\n[[\u2603 radio 1]]",
                        images: {},
                        widgets: {
                            "radio 1": {
                                graded: true,
                                version: {
                                    major: 1,
                                    minor: 0,
                                },
                                static: false,
                                type: "radio",
                                options: {
                                    choices: [
                                        {
                                            id: "0-0-0-0-0",
                                            content: "Hola",
                                            isNoneOfTheAbove: false,
                                            correct: true,
                                            rationale:
                                                "The Spanish-speaking countries typically say Hola.",
                                        },
                                        {
                                            id: "1-1-1-1-1",
                                            content: "Hey",
                                            isNoneOfTheAbove: false,
                                            correct: true,
                                            rationale:
                                                "This is used to attract someone's attention.",
                                        },
                                        {
                                            id: "2-2-2-2-2",
                                            content: "Hi",
                                            isNoneOfTheAbove: false,
                                            correct: true,
                                            rationale:
                                                "This is used as friendly greeting.",
                                        },
                                        {
                                            id: "3-3-3-3-3",
                                            content: "Goodbye",
                                            isNoneOfTheAbove: false,
                                            correct: false,
                                            rationale:
                                                "Some people like to say Goodbye.",
                                        },
                                        {
                                            id: "4-4-4-4-4",
                                            content: "None of these",
                                            isNoneOfTheAbove: true,
                                            correct: false,
                                        },
                                    ],
                                    hasNoneOfTheAbove: true,
                                    multipleSelect: true,
                                    randomize: false,
                                    deselectEnabled: false,
                                },
                                alignment: "default",
                            },
                        },
                        hint: {
                            content: "This is an example hint.",
                            images: {},
                            widgets: {},
                        },
                        widgetEnabled: true,
                        immutableWidgets: false,
                    },
                ],
            },
            version: {major: 0, minor: 0},
        },
    },
};

export const groupedRadioQuestion: PerseusRenderer = {
    content: "---\n\n##Check your understanding!\n\n[[☃ graded-group 1]]\n\n",
    images: {},
    widgets: {
        "graded-group 1": {
            type: "graded-group",
            alignment: "default",
            static: false,
            graded: true,
            options: {
                title: "Metabolic strategies of bacteria",
                content:
                    "Which of the following values of $x$ satisfies the equation $\\sqrt{64}=x$ ?\n\n[[\u2603 radio 1]]\n\n",
                images: {},
                widgets: {
                    "radio 1": {
                        graded: true,
                        version: {
                            major: 1,
                            minor: 0,
                        },
                        static: false,
                        type: "radio",
                        options: {
                            choices: [
                                {
                                    id: "1-1-1-1-1",
                                    content: "$-8$ and $8$",
                                    correct: false,
                                    rationale:
                                        "This is not the correct answer.",
                                },
                                {
                                    id: "2-2-2-2-2",
                                    content: "$-8$",
                                    correct: false,
                                    rationale:
                                        "This is not the correct answer.",
                                },
                                {
                                    id: "3-3-3-3-3",
                                    content: "$8$",
                                    correct: true,
                                    isNoneOfTheAbove: false,
                                    rationale: "This is the correct answer.",
                                },
                                {
                                    id: "4-4-4-4-4",
                                    content:
                                        "No value of $x$ satisfies the equation.",
                                    correct: false,
                                    isNoneOfTheAbove: false,
                                    rationale:
                                        "This is not the correct answer.",
                                },
                            ],
                            countChoices: false,
                            hasNoneOfTheAbove: false,
                            multipleSelect: false,
                            randomize: false,
                            deselectEnabled: false,
                        },
                        alignment: "default",
                    },
                },
                hint: {
                    content: "This is an example hint.",
                    images: {},
                    widgets: {},
                },
            },
            version: {major: 0, minor: 0},
        },
    },
};
