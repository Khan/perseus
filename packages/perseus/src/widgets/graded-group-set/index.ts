export {default} from "./graded-group-set";
