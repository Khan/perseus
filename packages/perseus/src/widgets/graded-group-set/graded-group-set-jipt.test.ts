/**
 * These GradedGroupSet tests are specifically for the JIPT integration.
 * This widget renders differently when JIPT is active to make translation
 * easier for translators.
 */

import {testDependencies} from "../../../../../testing/test-dependencies";
import * as Dependencies from "../../dependencies";
import {renderQuestion} from "../__testutils__/renderQuestion";

import {article1} from "./graded-group-set.testdata";

describe("graded-group-set", () => {
    beforeEach(() => {
        jest.spyOn(Dependencies, "getDependencies").mockReturnValue({
            ...testDependencies,
            JIPT: {
                useJIPT: true,
            },
        });
    });

    it("should render all graded groups", () => {
        // Arrange and Act
        const {container} = renderQuestion(article1);

        // Assert
        expect(container).toMatchSnapshot();
    });
});
