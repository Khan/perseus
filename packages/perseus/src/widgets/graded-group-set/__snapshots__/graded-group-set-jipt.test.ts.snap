// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graded-group-set should render all graded groups 1`] = `
<div>
  <div
    class="perseus-renderer perseus-renderer-responsive"
  >
    <div
      class="paragraph"
      data-perseus-paragraph-index="0"
    >
      <h1>
        Section 1: Adding tenths less than one
      </h1>
    </div>
    <div
      class="paragraph"
      data-perseus-paragraph-index="1"
    >
      <div
        class="paragraph"
      >
        <div
          class="perseus-widget-container widget-nohighlight widget-block"
        >
          <div
            class="container_14sgdtb"
          >
            <div
              class="perseus-graded-group"
            >
              <div
                class="title_1lagzkb"
              >
                Problem 1a
              </div>
              <div
                class="perseus-renderer perseus-renderer-responsive"
              >
                <div
                  class="paragraph"
                  data-perseus-paragraph-index="0"
                >
                  <div
                    class="paragraph"
                  >
                    <span
                      style="white-space: nowrap;"
                    >
                      <span />
                      <span
                        class="mock-TeX"
                      >
                        0.5 + 0.4 =
                      </span>
                      <span />
                    </span>
                       
                    <div
                      class="perseus-widget-container widget-nohighlight widget-inline-block"
                    >
                      <input
                        aria-disabled="false"
                        aria-invalid="false"
                        aria-required="false"
                        autocapitalize="off"
                        autocomplete="off"
                        autocorrect="off"
                        class="input_1ck1z8k-o_O-LabelMedium_1rew30o-o_O-default_1tc7wjm-o_O-defaultFocus_lrnqlz-o_O-inputWithExamples_1y6ajxo"
                        id=":r0:"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                      <span
                        id="aria-for-:r0:"
                        style="display: none;"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <p
                aria-live="polite"
                role="status"
              />
              <button
                aria-disabled="false"
                class="button_vr44p2-o_O-shared_lwskrm-o_O-default_ggcqzw"
                role="button"
                type="button"
              >
                <span
                  class="text_f1191h-o_O-LabelLarge_5s82ln-o_O-text_1awp0u1"
                >
                  Check
                </span>
              </button>
              <button
                class="showHintLink_oak3yy"
                tabindex="0"
              >
                Explain
              </button>
            </div>
            <div
              class="perseus-graded-group"
            >
              <div
                class="title_1lagzkb"
              >
                Problem 1b
              </div>
              <div
                class="perseus-renderer perseus-renderer-responsive"
              >
                <div
                  class="paragraph"
                  data-perseus-paragraph-index="0"
                >
                  <div
                    class="paragraph"
                  >
                    <span
                      style="white-space: nowrap;"
                    >
                      <span />
                      <span
                        class="mock-TeX"
                      >
                        0.6 + 0.4 =
                      </span>
                      <span />
                    </span>
                       
                    <div
                      class="perseus-widget-container widget-nohighlight widget-inline-block"
                    >
                      <input
                        aria-disabled="false"
                        aria-invalid="false"
                        aria-required="false"
                        autocapitalize="off"
                        autocomplete="off"
                        autocorrect="off"
                        class="input_1ck1z8k-o_O-LabelMedium_1rew30o-o_O-default_1tc7wjm-o_O-defaultFocus_lrnqlz-o_O-inputWithExamples_1y6ajxo"
                        id=":r3:"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                      <span
                        id="aria-for-:r3:"
                        style="display: none;"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <p
                aria-live="polite"
                role="status"
              />
              <button
                aria-disabled="false"
                class="button_vr44p2-o_O-shared_lwskrm-o_O-default_ggcqzw"
                role="button"
                type="button"
              >
                <span
                  class="text_f1191h-o_O-LabelLarge_5s82ln-o_O-text_1awp0u1"
                >
                  Check
                </span>
              </button>
              <button
                class="showHintLink_oak3yy"
                tabindex="0"
              >
                Explain
              </button>
            </div>
            <div
              class="perseus-graded-group"
            >
              <div
                class="title_1lagzkb"
              >
                Problem 1c
              </div>
              <div
                class="perseus-renderer perseus-renderer-responsive"
              >
                <div
                  class="paragraph"
                  data-perseus-paragraph-index="0"
                >
                  <div
                    class="paragraph"
                  >
                    <span
                      style="white-space: nowrap;"
                    >
                      <span />
                      <span
                        class="mock-TeX"
                      >
                        0.8 + 0.4 =
                      </span>
                      <span />
                    </span>
                       
                    <div
                      class="perseus-widget-container widget-nohighlight widget-inline-block"
                    >
                      <input
                        aria-disabled="false"
                        aria-invalid="false"
                        aria-required="false"
                        autocapitalize="off"
                        autocomplete="off"
                        autocorrect="off"
                        class="input_1ck1z8k-o_O-LabelMedium_1rew30o-o_O-default_1tc7wjm-o_O-defaultFocus_lrnqlz-o_O-inputWithExamples_1y6ajxo"
                        id=":r6:"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                      <span
                        id="aria-for-:r6:"
                        style="display: none;"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <p
                aria-live="polite"
                role="status"
              />
              <button
                aria-disabled="false"
                class="button_vr44p2-o_O-shared_lwskrm-o_O-default_ggcqzw"
                role="button"
                type="button"
              >
                <span
                  class="text_f1191h-o_O-LabelLarge_5s82ln-o_O-text_1awp0u1"
                >
                  Check
                </span>
              </button>
              <button
                class="showHintLink_oak3yy"
                tabindex="0"
              >
                Explain
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="paragraph"
      data-perseus-paragraph-index="2"
    >
      <div
        class="paragraph"
      >
        Beautiful, let's move on to problems with whole numbers and tenths.
      </div>
    </div>
  </div>
</div>
`;
